indent/handlebars.vim

```diff
@@ -79,16 +79,58 @@ function! GetHandlebarsIndent(...)
   " all indent rules only apply if the block opening/closing
   " tag is on a separate line
 
-  " indent after block {{#block
-  if prevLine =~# '\v\s*\{\{[#^].*\s*'
+  " " check for a hanging attribute
+  let lastLnumCol = col([lnum, '$']) - 1
+  if synIDattr(synID(lnum, lastLnumCol, 1), "name") =~ '^mustache'
+        \ && prevLine !~# '}}\s*$'
+    let hangingAttributePattern = '{{[#^]\=\%(\k\|[/-]\)\+\s\+\zs\k\+='
+    let standaloneComponentPattern = '^\s*{{\%(\k\|[/-]\)\+\s*$'
+
+    if prevLine =~ hangingAttributePattern
+      " {{component attribute=value
+      "             other=value}}
+      let [line, col] = searchpos(hangingAttributePattern, 'Wbn', lnum)
+      if line == lnum
+        return col - 1
+      endif
+    elseif prevLine =~ standaloneComponentPattern
+      " {{component
+      "   attribute=value}}
+      return indent(lnum) + sw
+    endif
+  endif
+
+  let componentClosingPattern = '\v^\s*\{\{\/\S*\}\}\s*'
+
+  " check for a closing }}, indent according to the opening one
+  let saved_pos = getpos('.')
+  if prevLine =~# '}}$' && prevLine !~# '^\s*{{' &&
+        \ currentLine !~# componentClosingPattern &&
+        \ search('}}$', 'Wb')
+    let [line, col] = searchpairpos('{{', '', '}}', 'Wb')
+    if line > 0
+      if strpart(getline(line), col - 1, 3) =~ '{{[#^]'
+        " then it's a block component, indent a shiftwidth more
+        return indent(line) + sw
+      else
+        return indent(line)
+      endif
+    else
+      call setpos('.', saved_pos)
+    endif
+  endif
+
+  " indent after block: {{#block, {{^block
+  if prevLine =~# '\v\s*\{\{[#^].*\s*' &&
+        \ prevLine !~# '{{[#^]\(.\{}\)\s.\{}}}.*{{\/\1}}'
```

Comment:
Is this additional check separate from the purpose of this PR? This looks unrelated to hanging attributes to me but I might be misreading it.

Reply:
I think it's not related. I guess I've added it to avoid increasing indent for block components on the same line, but the lines below seem to take care of that.

I removed the check locally, and it seems like it doesn't affect anything. I'll commit a removal of it, and if you see any regressions, we can put it back, I guess.

Reply:
Thanks -- if it's a useful change, I'd love to see it in a separate PR